sessionize/lib/src/commonMain/kotlin/co/touchlab/sessionize/AppContext.kt

```diff
@@ -1,188 +1,52 @@
-import co.touchlab.sessionize.platform.reminderNotificationsEnabled
-import co.touchlab.stately.concurrency.ThreadLocalRef
-import co.touchlab.stately.concurrency.value
-import kotlinx.coroutines.launch
 import kotlinx.serialization.json.Json
-import kotlin.coroutines.CoroutineContext
 
 object AppContext {
-
     private val primeJson = Json.nonstrict
-    val dbHelper = SessionizeDbHelper()
-
-    val KEY_FIRST_RUN = "FIRST_RUN1"
-    val KEY_LAST_LOAD = "LAST_LOAD"
-    val USER_UUID = "USER_UUID"
-    val FEEDBACK_ENABLED = "FEEDBACK_ENABLED"
-    val REMINDERS_ENABLED = "REMINDERS_ENABLED"
-    val LOCAL_NOTIFICATIONS_ENABLED = "LOCAL_NOTIFICATIONS_ENABLED"
-    val TWO_HOURS_MILLIS = 2 * 60 * 60 * 1000
-    val TEN_MINS_MILLIS = 1000 * 10 * 60
-
-
-    fun initAppContext() {
-        dbHelper.initDatabase(ServiceRegistry.dbDriver)
-
-        ServiceRegistry.notificationsApi.initializeNotifications{success ->
-
-            backToFront({success},{
-                if(it) createNotificationsForSessions() else cancelNotificationsForSessions()
-            })
-
-        }
-    }
-
-    internal val sessionQueries: SessionQueries
-        get() = AppContext.dbHelper.instance.sessionQueries
-
-    internal val userAccountQueries: UserAccountQueries
-        get() = AppContext.dbHelper.instance.userAccountQueries
-
-    internal val roomQueries: RoomQueries
-        get() = AppContext.dbHelper.instance.roomQueries
-
-    internal val sponsorQueries: SponsorQueries
-        get() = AppContext.dbHelper.instance.sponsorQueries
-
-    val staticFileLoader: (filePrefix: String, fileType: String) -> String?
-        get() = ServiceRegistry.staticFileLoader
-
-    val clLogCallback: (s: String) -> Unit
-        get() = ServiceRegistry.clLogCallback
-
-    private fun firstRun(): Boolean = ServiceRegistry.appSettings.getBoolean(KEY_FIRST_RUN, true)
-
-    private fun updateFirstRun() {
-        ServiceRegistry.appSettings.putBoolean(KEY_FIRST_RUN, false)
-    }
-
-    fun userUuid(): String {
-        if (ServiceRegistry.appSettings.getString(USER_UUID).isBlank()) {
-            ServiceRegistry.appSettings.putString(USER_UUID, createUuid())
-        }
-        return ServiceRegistry.appSettings.getString(USER_UUID)
-    }
-
-    fun loadSponsors(): String? {
-        return staticFileLoader("sponsors", "json")
-    }
-
-    fun loadSpeakers(): String? {
-        return staticFileLoader("speakers", "json")
-    }
 
-    fun loadSchedule(): String? {
-        return staticFileLoader("schedule", "json")
-    }
-
-    fun loadAbout(): String? {
-        return staticFileLoader.invoke("about", "json")
-    }
-
-    //Split these up so they can individually succeed/fail
-    fun dataLoad() {
-        if (firstRun()) {
-            backgroundTask({
-                try {
-                    seedFileLoad()
-                } catch (e: Exception) {
-                    logException(e)
+    fun initAppContext(networkRepo: NetworkRepo = NetworkRepo,
```

Comment:
This is somewhat funky. The parameters are available so that we can pass in mocks when testing, but default to the singleton objects otherwise. So normal usage doesn't need to worry about it. I haven't written the tests yet because without a mocking library it's a PITA and I didn't want to inflate this PR further

Reply:
K/N doesn't handle default params, unfortunately, so you do have to add them to usage there :(